### PR TITLE
feat: introduces new user fields in titkok ads

### PR DESCRIPTION
**Pull request by anantjain45823**
## What are the changes introduced in this PR?
This PR introduces new fields introduced for user object in tiktok ads refer [this](https://business-api.tiktok.com/portal/docs?id=1771101151059969#item-link-Web%20Standard%20Events) documentation for reference 

Write a brief explainer on your code changes.

## What is the related Linear task?

Resolves INT-2411

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [ ] My code follows the style guidelines of this project

- [ ] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [ ] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added in **new readability format**?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3575/test-report.html

**Review by ItsSudip — DISMISSED** [review]

File: src/v0/destinations/tiktok_ads/data/TikTokTrackV2.json [hunk 1]
```diff
@@ -126,5 +126,53 @@
...
+      "context.traits.postalCode",
+      "context.traits.address.zip",
+      "context.traits.address.zip_code",
+      "context.traits.address.postalcode",
+      "context.traits.address.postalCode",
+      "properties.zip",
+      "properties.postalcode",
+      "properties.postalCode",
+      "properties.zip_code"
+    ],
```
Comment (sanpj2292) [inline src/v0/destinations/tiktok_ads/data/TikTokTrackV2.json:173]: We already have generic mapping right ?! Would that not suffice ?

Comment (sanpj2292) [reply src/v0/destinations/tiktok_ads/data/TikTokTrackV2.json:173]: src/v0/util/data/GenericFieldMapping.json

Comment (sanpj2292) [reply src/v0/destinations/tiktok_ads/data/TikTokTrackV2.json:173]: Check if we can use genericMapping for city, state, country fields

Comment (anantjain45823) [reply src/v0/destinations/tiktok_ads/data/TikTokTrackV2.json]: changed for zipcode and others won't suffice

**Review by sanpj2292 — APPROVED** [review]

File: src/v0/destinations/tiktok_ads/data/TikTokTrackV2.json [hunk 2]
```diff
@@ -126,5 +126,41 @@
...
+    "destKey": "user.city",
+    "sourceKeys": ["context.traits.city", "context.traits.address.city", "properties.city"]
+  },
+  {
+    "destKey": "user.country",
+    "sourceKeys": ["context.traits.country", "context.traits.address.country", "properties.country"]
+  },
+  {
+    "destKey": "user.state",
+    "sourceKeys": ["context.traits.address.state", "context.traits.state", "properties.state"]
```
Comment (sanpj2292) [inline src/v0/destinations/tiktok_ads/data/TikTokTrackV2.json:156]: 1. Can you check order and compare with other fields like city, country etc ?
2. Should we also add `context.traits.address.state` as one of the sourceKeys ?

Comment (anantjain45823) [reply src/v0/destinations/tiktok_ads/data/TikTokTrackV2.json:156]: These are user properties so I guess we should be ok with these fields

**Review by ItsSudip — APPROVED** [review]

**Pull request merged**